unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/QuestsTrackerHUD/QuestsTrackerHUDView.cs

```diff
@@ -0,0 +1,174 @@
+                questEntry.OnLayoutRebuildRequested += () => layoutRebuildRequested = true;
+                questEntry.Populate(quest);
+                currentEntries.Add(quest.id, questEntry);
+            }
+
+            RefreshLastUpdateTime(quest.id, isPinned);
+            questEntry.transform.SetSiblingIndex(0);
+
+            questEntry.Populate(quest);
+            questEntry.SetPinStatus(isPinned);
+            layoutRebuildRequested = true;
+        }
+
+        public void RemoveEntry(string questId)
+        {
+            questsToBeAdded.Remove(questId);
+            if (!currentEntries.TryGetValue(questId, out QuestsTrackerEntry entry))
+                return;
+
+            currentEntries.Remove(questId);
+            lastUpdateTimestamp.Remove(questId);
+            Destroy(entry.gameObject);
+        }
+
+        public void PinQuest(string questId)
+        {
+            if (currentEntries.TryGetValue(questId, out QuestsTrackerEntry entry))
+            {
+                entry.SetPinStatus(true);
+                RefreshLastUpdateTime(questId, true);
+            }
+            else
+            {
+                if (questsToBeAdded.Contains(questId))
+                    return;
+
+                questsToBeAdded.Add(questId);
+            }
+        }
+
+        public void UnpinQuest(string questId)
+        {
+            if (!currentEntries.TryGetValue(questId, out QuestsTrackerEntry entry))
+                return;
+
+            entry.SetPinStatus(false);
+            RefreshLastUpdateTime(questId, false);
+        }
+
+        private void Update()
+        {
+            if (layoutRebuildRequested)
+            {
+                layoutRebuildRequested = false;
+                Utils.ForceRebuildLayoutImmediate(questsContainer);
+            }
+
+            for (int i = 0; i < ENTRIES_PER_FRAME && questsToBeAdded.Count > 0; i++)
+            {
+                string questId = questsToBeAdded.First();
+                questsToBeAdded.RemoveAt(0);
+                AddOrUpdateQuest(questId, pinnedQuests.Contains(questId));
+            }
+        }
+
+        internal void RefreshLastUpdateTime(string questId, bool isPinned)
+        {
+            DateTime dateToSet = isPinned ? DateTime.MaxValue : DateTime.Now;
+
+            if (lastUpdateTimestamp.ContainsKey(questId))
+                lastUpdateTimestamp[questId] = dateToSet;
+            else
+                lastUpdateTimestamp.Add(questId, dateToSet);
+        }
+
+        public void ClearEntries()
+        {
+            lastUpdateTimestamp.Clear();
+            foreach ((string key, QuestsTrackerEntry value) in currentEntries)
+            {
+                Destroy(value.gameObject);
+            }
+            currentEntries.Clear();
+        }
+
+        public void SetVisibility(bool visibility)
+        {
+            gameObject.SetActive(visibility);
+        }
+
+        private IEnumerator DispatchEntriesRoutine()
```

Comment:
Here we are using `Update` to add entries and a coroutine to remove them. I think we should unify the approach.

Reply:
Makes sense!